### PR TITLE
feat: sbtc sponsor swap fees

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out Leather build 0417c9b — [Extension build](https://github.com/leather-io/extension/actions/runs/12341631889), [Test report](https://leather-io.github.io/playwright-reports/feat/sbtc-sponsor-swap-fees), [Storybook](https://feat/sbtc-sponsor-swap-fees--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=feat/sbtc-sponsor-swap-fees)

@alexp3y maybe a start in trying to use the sponsor fees api?

### CODE REVIEW

File: src/app/pages/swap/hooks/use-sponsor-tx-fees.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    [
+      filteredBalanceQuery.data?.availableBalance.amount,
+      isSbtcSwapsEnabled,
+      setIsIdle,
+      verificationResult?.isEligible,
+    ]
+  );
+
+  const submitSponsoredTx = useCallback(async () => {
+    if (isEligibleForSponsor) {
```
Comment (kyranjamie) [inline src/app/pages/swap/hooks/use-sponsor-tx-fees.tsx:79]: Suggest early return here

**Review by kyranjamie — COMMENTED** [review]
Loads of work here @fbwoolf! Let me know how I can test this

File: src/app/pages/swap/hooks/use-sponsor-tx-fees.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      })
+        .then(result => {
+          setVerificationResult(result);
+        })
+        .catch(e => {
+          logger.error('Verification failure: ', e);
+          setVerificationResult({ isEligible: false });
+        })
+        .finally(() => {
+          setIsIdle;
```
Comment (kyranjamie) [inline src/app/pages/swap/hooks/use-sponsor-tx-fees.tsx:58]: is this supposed to be invoked?

File: src/app/query/sbtc/sponsored-transactions.query.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  sponsoredTx: StacksTransaction
+): Promise<SbtcSponsorshipSubmissionResult> {
+  logger.debug('Submitting Sponsored sBTC Transaction!');
+  try {
+    const { data } = await axios.post('http://localhost:5001/submit', {
+      tx: bytesToHex(sponsoredTx.serialize()),
+    });
+    logger.debug('sBTC Sponsorship Success:', data.txId);
+    return {
+      txId: data.txId,
```
Comment (kyranjamie) [inline src/app/query/sbtc/sponsored-transactions.query.ts:42]: `txid` for consistency with elsewhere in code🙏🏼

File: src/app/pages/swap/hooks/use-sponsor-tx-fees.tsx (hunk 2 above)
Comment (fbwoolf) [reply src/app/pages/swap/hooks/use-sponsor-tx-fees.tsx:58]: Yes, haha.

File: src/app/query/sbtc/sponsored-transactions.query.ts (hunk 3 above)
Comment (fbwoolf) [reply src/app/query/sbtc/sponsored-transactions.query.ts:42]: @alexp3y

**Comment by camerow** [comment]
Closing in favor of the more updated #6012

**Pull request closed without merging**